src/models/timeaxisheaderview.cpp

```diff
@@ -0,0 +1,109 @@
+#include "timeaxisheaderview.h"
+
+#include <QDebug>
+#include <QPainter>
+#include <QtMath>
+
+#include "eventmodel.h"
+#include "filterandzoomstack.h"
+#include "../util.h"
+
+#include "../../3rdparty/PrefixTickLabels/src/PrefixTickLabels.h"
+
+TimeAxisHeaderView::TimeAxisHeaderView(FilterAndZoomStack* filterAndZoomStack, QWidget* parent)
+    : QHeaderView(Qt::Horizontal, parent)
+    , m_filterAndZoomStack(filterAndZoomStack)
+{
+    setMinimumHeight(3*fontMetrics().height() + s_tickHeight);
+    setStretchLastSection(true);
+
+    auto emitHeaderDataChanged = [this]() { emit headerDataChanged(this->orientation(), EventModel::EventsColumn, EventModel::EventsColumn); };
+    connect(filterAndZoomStack, &FilterAndZoomStack::filterChanged, this, emitHeaderDataChanged);
+    connect(filterAndZoomStack, &FilterAndZoomStack::zoomChanged, this, emitHeaderDataChanged);
+}
+
+void TimeAxisHeaderView::setTimeRange(const Data::TimeRange& timeRange)
+{
+    m_timeRange = timeRange;
+    emit headerDataChanged(orientation(), 1, 1);
+}
+
+void TimeAxisHeaderView::paintSection(QPainter* painter, const QRect& rect, int logicalIndex) const
+{
+    if (painter == nullptr)
+        return;
+
+    painter->save();
+    QHeaderView::paintSection(painter, rect, logicalIndex);
+    painter->restore();
+    if (logicalIndex != EventModel::EventsColumn)
+        return;
+
+    auto zoomTime = m_filterAndZoomStack->zoom().time;
+    if (!zoomTime.isValid()) {
+        zoomTime = m_timeRange; // full
+    }
+    if (!zoomTime.isValid()) {
+        return; // bailing out, no valid range to show
+    }
+    const double oneNanoSecond = 1.0e-9;
+    const double start = (zoomTime.start - m_timeRange.start) * oneNanoSecond;
+    const double end = (zoomTime.end - m_timeRange.start) * oneNanoSecond;
+
+    const double resolution = (end - start) / rect.width();
+    const auto xForTime = [rect, start, resolution](const double time) {
+        return rect.x() + qRound((time - start) / resolution);
+    };
+
+    const int fontSize = painter->fontMetrics().height();
+    const int startY = rect.height() - s_tickHeight - 2*fontSize;
+    const int endLabelWidth = painter->fontMetrics().horizontalAdvance(QLatin1String("-xXXXm"));
+    const int targetNbTicks = rect.width() / endLabelWidth;
+    const PrefixTickLabels pfl(start, end, targetNbTicks);
+
+    const QColor tickColor = palette().windowText().color();
```

Comment:
I assume these lines here lead to the broken colors on my dark color scheme. could you please use something from `KColorScheme` instead of trying to create your own color via darkening?

Reply:
I removed the darkening, sticking to the colours of the pallets. Does that fix the issue in your color scheme?